src/main/java/org/openstreetmap/atlas/utilities/command/subcommands/AtlasDiffCommand.java

```diff
@@ -272,10 +247,165 @@ public void registerOptionsAndArguments()
+                FOLDER_SEARCH_RECURSIVE_OPTION_DESCRIPTION, OptionOptionality.OPTIONAL,
+                FULL_CONTEXT, GEOJSON_CONTEXT, LDGEOJSON_CONTEXT);
         registerArgument(BEFORE_ATLAS_ARGUMENT, ArgumentArity.UNARY, ArgumentOptionality.REQUIRED,
                 DEFAULT_CONTEXT, LDGEOJSON_CONTEXT, GEOJSON_CONTEXT, FULL_CONTEXT);
         registerArgument(AFTER_ATLAS_ARGUMENT, ArgumentArity.UNARY, ArgumentOptionality.REQUIRED,
                 DEFAULT_CONTEXT, LDGEOJSON_CONTEXT, GEOJSON_CONTEXT, FULL_CONTEXT);
         super.registerOptionsAndArguments();
     }
+
+    boolean compute(final AtlasDiffCommandContext context, final File beforeAtlasFile,
+            final File afterAtlasFile)
+    {
+        boolean result = false;
+        final Map<String, File> beforeNamesToFiles = new HashMap<>();
+        final Map<String, File> afterNamesToFiles = new HashMap<>();
+        final List<File> beforeFilesToConsider = context.isRecursive()
+                ? beforeAtlasFile.listFilesRecursively(false)
+                : beforeAtlasFile.listFiles(false);
+        final List<File> afterFilesToConsider = context.isRecursive()
+                ? afterAtlasFile.listFilesRecursively(false)
+                : afterAtlasFile.listFiles(false);
+        beforeFilesToConsider.stream().filter(FileSuffix.ATLAS::matches).forEach(
+                file -> beforeNamesToFiles.put(getRelativeFileName(beforeAtlasFile, file), file));
+        afterFilesToConsider.stream().filter(FileSuffix.ATLAS::matches).forEach(
+                file -> afterNamesToFiles.put(getRelativeFileName(afterAtlasFile, file), file));
+        final Set<String> filesOnlyInBefore = Sets.difference(beforeNamesToFiles.keySet(),
+                afterNamesToFiles.keySet());
+        final Set<String> filesOnlyInAfter = Sets.difference(afterNamesToFiles.keySet(),
+                beforeNamesToFiles.keySet());
+        final Set<String> filesInBoth = Sets.intersection(beforeNamesToFiles.keySet(),
+                afterNamesToFiles.keySet());
+        if (!filesOnlyInBefore.isEmpty())
+        {
+            final String warnMessage = "Files only in Before Atlas folder:";
+            this.outputDelegate.printlnWarnMessage(warnMessage);
+            filesOnlyInBefore.stream().sorted().forEach(this.outputDelegate::printlnWarnMessage);
+            result = true;
+        }
+        if (!filesOnlyInAfter.isEmpty())
+        {
+            final String warnMessage = "Files only in After Atlas folder:";
+            this.outputDelegate.printlnWarnMessage(warnMessage);
+            filesOnlyInAfter.stream().sorted().forEach(this.outputDelegate::printlnWarnMessage);
+            result = true;
+        }
+        for (final String name : filesInBoth.stream().sorted().collect(Collectors.toList()))
+        {
+            final Atlas beforeAtlas = load(beforeNamesToFiles.get(name));
+            final Atlas afterAtlas = load(afterNamesToFiles.get(name));
```

Comment:
If you make `load` call `AtlasResourceLoader#safeLoad`, then you can have optionals here and just skip files that were not atlases. Again, this would automatically handle loading any kind of atlas supported by `AtlasResourceLoader`, so if in the future we add more loading capabilities, this command would automatically benefit.

Reply:
So to be able to compare the Atlas files missing in each folder, I am using the other option with the checkAtlas in your other comment..